blockly-core/core/ui/function_editor.js

```diff
@@ -798,6 +799,20 @@ Blockly.FunctionEditor.prototype.addDeleteButton_ = function () {
       this.onDeletePressed.bind(this));
 };
 
+/**
+ * Show or hide delete button. If none exists and it should be shown, create
+ * it.
+ */
+Blockly.FunctionEditor.prototype.setDeleteButtonVisible_ = function(visible) {
+  if (this.deleteButton_) {
+    this.deleteButton_.setVisible(visible);
+  } else if (visible) {
+    this.hasDeleteButton = true;
```

Comment:
Yeah, turns out `hasDeleteButton` was no longer needed. It was just being used to differentiate between the function editor and the contract editor (showing the delete button only in the latter), but they now have the same behavior.

Removed it.